### PR TITLE
[14.0][IMP] sale_commission_agent_restrict: hide customers contacts (…

### PR DESCRIPTION
**Pull request by ilyasProgrammer**
…do not populate agent_ids among child ids of the customer)

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
Functional ok

@ilyasProgrammer squash commits and set ready for review

**Review by elvise — APPROVED** [review]
LGTM!

**Comment by francesco-ooops** [comment]
@OCA/crm-sales-marketing-maintainers merge?

File: sale_commission_agent_restrict/models/res_partner.py [hunk 1]
```diff
@@ -35,3 +35,18 @@ def create(self, vals):
             vals["agent_ids"] = [(4, self.env.user.partner_id.id, 0)]
         res = super(ResPartner, self).create(vals)
         return res
+
+    @api.model
+    def _commercial_fields(self):
+        res = super()._commercial_fields()
+        for agent in self.agent_ids:
```
Comment (rousseldenis) [inline sale_commission_agent_restrict/models/res_partner.py]: @ilyasProgrammer This is not correct as function is a model decorated one.

You should find another way.

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

**Comment by ilyasProgrammer** [comment]
@rousseldenis updated

**Comment by francesco-ooops** [comment]
> @rousseldenis updated

functional ok

**Comment by francesco-ooops** [comment]
@rousseldenis good now? :)

File: sale_commission_agent_restrict/models/res_partner.py [hunk 2]
```diff
@@ -35,3 +35,19 @@ def create(self, vals):
             vals["agent_ids"] = [(4, self.env.user.partner_id.id, 0)]
         res = super(ResPartner, self).create(vals)
         return res
+
+    def _update_fields_values(self, fields):
+        res = super(ResPartner, self)._update_fields_values(fields)
+        for agent in self.agent_ids:
+            user_id = self.env["res.users"].search([("partner_id", "=", agent.id)])
```
Comment (rousseldenis) [inline sale_commission_agent_restrict/models/res_partner.py]: That's better to use `user_id = agent.user_ids` as it benefits from auto_join and avoid to do a search per agent.

See https://github.com/odoo/odoo/blob/14.0/odoo/addons/base/models/res_partner.py#L219

Comment (ilyasProgrammer) [reply sale_commission_agent_restrict/models/res_partner.py]: ok

File: sale_commission_agent_restrict/models/sale.py [hunk 3]
```diff
@@ -18,3 +21,16 @@ def _compute_commission_total(self):
                 lambda x: x.agent_id == self.env.user.partner_id
             )
             record.current_agent_total = sum(my_com.mapped("amount"))
+
+    @api.depends("company_id")
```
Comment (rousseldenis) [inline sale_commission_agent_restrict/models/sale.py]: You should use : 

```suggestion
    @api.depends_context("company", "uid")
```

**Comment by francesco-ooops** [comment]
@rousseldenis ok now?

**Comment by francesco-ooops** [comment]
@rousseldenis gentle reminder :)

**Comment by francesco-ooops** [comment]
@rousseldenis could you take a look? thanks!

**Comment by francesco-ooops** [comment]
@rousseldenis gentle reminder

File: sale_commission_agent_restrict/models/res_partner.py (hunk 1 above)
Comment (simahawk) [reply sale_commission_agent_restrict/models/res_partner.py]: AFAIS is not decorated w/ `@api.model` https://github.com/odoo/odoo/blob/14.0/odoo/addons/base/models/res_partner.py#L405 and is - apparently  - called always on the right recordset.

File: sale_commission_agent_restrict/models/res_partner.py [hunk 4]
```diff
@@ -35,3 +35,19 @@ def create(self, vals):
             vals["agent_ids"] = [(4, self.env.user.partner_id.id, 0)]
         res = super(ResPartner, self).create(vals)
         return res
+
+    def _update_fields_values(self, fields):
+        res = super(ResPartner, self)._update_fields_values(fields)
+        for agent in self.agent_ids:
+            for user_id in agent.user_ids:
+                if (
+                    "agent_ids" in res.keys()
```
Comment (simahawk) [inline sale_commission_agent_restrict/models/res_partner.py]: `"agent_ids" in res` this check should be done before the main loop IMO.
Also, `user_id` is actually a `user`.

File: sale_commission_agent_restrict/models/sale.py [hunk 5]
```diff
@@ -9,6 +11,7 @@ class SaleOrder(models.Model):
         compute="_compute_commission_total",
         store=True,
     )
+    partner_id_domain = fields.Char(compute="_compute_partner_id_domain")
```
Comment (simahawk) [inline sale_commission_agent_restrict/models/sale.py:14]: can't we use a `Serialized` field instead? Just asking... I'm not sure it works w/ the domain widget ;)

File: sale_commission_agent_restrict/security/security.xml [hunk 6]
```diff
@@ -25,11 +25,10 @@
         <field name="model_id" ref="base.model_res_partner" />
         <field name="groups" eval="[(4,ref('group_agent_own_customers'))]" />
         <field name="domain_force">[
-            '|','|','|','|',
-            ('id', 'child_of', user.partner_id.child_ids.ids),
-            ('user_ids', 'in', user.company_id.user_ids.ids),
+            '|','|','|',
+            ('id', '=', user.partner_id.id),
+            ('user_ids', '=', user.id),
```
Comment (simahawk) [inline sale_commission_agent_restrict/security/security.xml]:
```suggestion
            ('user_ids', 'in', user.id),
```
seems more correct here

**Review by simahawk — CHANGES_REQUESTED** [review]
Additionally, can you please rewrite the commit?

1. should not include odoo version
2. should be shorter and have the description where it belongs: in the description of the commit

Thanks!

**Comment by ilyasProgrammer** [comment]
moved here https://github.com/OCA/commission/pull/382

**Pull request closed without merging**